docs/parameters.md

```diff
@@ -0,0 +1,322 @@
+      type: json
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: password
+```
+
+#### Other primitive types (int, bool, float)
+
+```yaml
+  ...
+  parameters:
+    - name: text
+      value: 1 # string by default, will be sent as "1"
+    - name: number
+      type: int
+      value: 1 # will be sent without quotes
+    - name: flag
+      type: bool
+      value: true
+    - name: fl
+      type: float
+      value: 123.5
+```
+
+#### Arrays
+A separate field `array: true` can be used to declare a value as an array. The
+element type is defined by the `type` value as described above.
+
+```yaml
+  ...
+  parameters:
+    - name: tags
+      array: true
+      value:
+      - foo
+      - bar
+    - name: jsonArray
+      type: json
+      array: true
+      value: [{"a": 1, "b": 2}]
+```
+
+### Passing all parameters from a single source at once
+
+In addition to support for explicitly specifying every parameter value fetched from 
+a particular source, Service Catalog also supports passing multiple parameters at 
+once using a `parametersFrom` field (similar to how `envFrom` works for `Pod`s).
+Instead of specifying a reference to a single `Secret` (or `ConfigMap`) key, you 
+just set the reference to the `Secret` (or `ConfigMap`), and all keys will be 
+automatically processed as separate parameters.
+
+In other words, instead of 
+```yaml
+  ...
+  parameters:
+    - name: key1
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key1
+    - name: key2
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key2
+     ...
+```
+you can just set it once in a `parametersFrom` field:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+You can also pass several `Secret`s and/or `ConfigMap`s at once:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: secret1
+    - secretRef:
+        name: secret2
+    - configMapRef:
+        name: configmap1
+```
+
+Parameter values could be passed as JSON objects, the same way as for [individual parameters](#json-object):
+```yaml
+  ...
+  parametersFrom:
+    - type: json
+      secretRef:
+        name: mysecret
+```
+
+### Conflict resolution
+
+When a parameter name exists in multiple sources defined in `parametersFrom` or `parameters` 
+field, the value associated with the last source will take precedence.
+Values defined in `parameters` with a duplicate name will take precedence over `parametersFrom`.
+This conflict resolution strategy is consistent with the one supported by Kubernetes
+for environment variables (`env` and `envFrom` fields) in `Pod` resource.
+
+For example, for the structure like this
+```yaml
+  ...
+  parameters:
+    - name: param
+      value: 1
+    - name: param
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key
+    - name: param
+      value: 2
+  parametersFrom:
+    - name: param
```

Comment:
@duglin I see what you mean, but I would prefer to keep it there:
1. name attribute is optional, and without it the result is "the entire set of params" at the top level
2. with the name specified it's still "the entire set of params", just not at the top level.
3. this feature is, again, based on the ideas from `envFrom`, where you can specify a prefix for all env vars taken from the secret. I thought that prefix is more awkward than nesting, but we can change it to the prefix if you prefer.